alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -150,26 +154,38 @@ class TransportDeleteWorkflowAction @Inject constructor(
                         }
                     }
 
-                    val deleteResponse = deleteWorkflow(workflow)
+                    val deleteResponse = deleteWorkflow(deleteRequest)
+                    var deleteWorkflowResponse = DeleteWorkflowResponse(deleteResponse.id, deleteResponse.version)
+
+                    val workflowMetadataId = WorkflowMetadata.getId(workflow.id)
+
+                    val metadataIdsToDelete = mutableListOf(workflowMetadataId)
+
                     if (deleteDelegateMonitors == true) {
-                        if (user != null && filterByEnabled) {
-                            // Un-stash the context
-                            withClosableContext(
```

Comment:
@eirsep @lezzago - if we conclude that we need to change the part I described in the comment above, we should definitely remove the context unstash so the program in next couple of lines, once we are deleting the monitor, can behave as an admin.

If we are going to keep the check in the comment above as it is, then the user must have monitor_delete cluster privilege - meaning that even before starting workflow deletion we would need to check it's cluster permission (which I think we shouldn't do and I don't know how to do since the security plugin does this I guess).

Reply:
@getsaurabh02 also, here

Reply:
@lezzago @eirsep @getsaurabh02 
Answered in previous response